tracee-ebpf/tracee/tracee.go

```diff
@@ -286,6 +286,23 @@ func New(cfg Config) (*Tracee, error) {
 		setEssential(VfsWritevEventID)
 	}
 
+	if t.eventsToTrace[SecuritySocketListenEventID] {
+		setEssential(ListenEventID)
+	}
+
+	if t.eventsToTrace[SecuritySocketConnectEventID] {
+		setEssential(ConnectEventID)
+	}
+
+	if t.eventsToTrace[SecuritySocketAcceptEventID] {
+		setEssential(AcceptEventID)
+		setEssential(Accept4EventID)
+	}
+
+	if t.eventsToTrace[SecuritySocketBindEventID] {
+		setEssential(BindEventID)
+	}
+
```

Comment:
Looking at your changes at tracepoint__raw_syscalls__sys_enter() again, I don't think that this is even necessary.
sockfd should be saved on sys_entry because shuold_trace() doesn't check if the event was chosen.

Can you please try to remove the above lines and check if sockfd is saved in the lsm hooks events when accept/connect/listen/bind is not chosen?

Reply:
you're right. it works without setting the syscalls as essential to the lsm hooks.